# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1202459352 18000
#      Fri Feb 08 03:29:12 2008 -0500
# Branch release-3-0-x
# Node ID eaaa74f38140f054168bcc00b715ddb4e32a9aab
# Parent  895a1c78828175e0a3982248135cf6424f7357a7
fix struct resizing bug

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2008-02-08  John W. Eaton  <jwe@octave.org>
+
+	* ov-struct.cc (octave_struct::subsref): Allow Cell::index to resize.
+
 2008-02-07  David Bateman  <dbateman@free.fr>
 
 	* ov-range.h (octave_range::sort): New functions.
 
 2008-01-30  Thomas Weber  <thomas.weber.mail@gmail.com>
 
 	* variables.cc (Fclear): Doc fix.
 
diff --git a/src/ov-struct.cc b/src/ov-struct.cc
--- a/src/ov-struct.cc
+++ b/src/ov-struct.cc
@@ -113,28 +113,28 @@ octave_struct::subsref (const std::strin
 	  {
 	    std::list<octave_value_list>::const_iterator p = idx.begin ();
 	    octave_value_list key_idx = *++p;
 
 	    Cell tmp = dotref (key_idx);
 
 	    if (! error_state)
 	      {
-		Cell t = tmp.index (idx.front ());
+		Cell t = tmp.index (idx.front (), true);
 
 		retval(0) = (t.length () == 1) ? t(0) : octave_value (t, true);
 
 		// We handled two index elements, so tell
 		// next_subsref to skip both of them.
 
 		skip++;
 	      }
 	  }
 	else
-	  retval(0) = map.index (idx.front ());
+	  retval(0) = map.index (idx.front (), true);
       }
       break;
 
     case '.':
       {
 	if (map.numel() > 0)
 	  {
 	    Cell t = dotref (idx.front ());
